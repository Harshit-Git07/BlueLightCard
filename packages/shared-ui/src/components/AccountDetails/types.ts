export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
