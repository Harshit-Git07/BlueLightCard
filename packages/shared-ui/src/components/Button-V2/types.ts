import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
