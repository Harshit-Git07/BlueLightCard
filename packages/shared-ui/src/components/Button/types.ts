import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
