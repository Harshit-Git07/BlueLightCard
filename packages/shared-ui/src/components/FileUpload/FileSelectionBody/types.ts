export interface FileSelectionProps {
  onFilesSelected: (files: File[]) => void;
  disabled?: boolean;
  allowedFileTypes?: string[]; // e.g. ['image/png', 'image/jpeg']
  maxFileSize?: number;
}
