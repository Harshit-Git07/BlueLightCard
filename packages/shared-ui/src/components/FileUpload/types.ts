export enum UploadStatus {
  Ready = 'ready',
  Loading = 'loading',
  Success = 'success',
  Error = 'error',
}

interface FileInfoBase {
  id: string;
  uploadStatus: UploadStatus;
  file: File;
}

interface FileInfoWaiting extends FileInfoBase {
  uploadStatus: UploadStatus.Ready | UploadStatus.Loading;
}

interface FileInfoUploadSuccess extends FileInfoBase {
  uploadStatus: UploadStatus.Success;
  code?: number;
}

interface FileInfoUploadError extends FileInfoBase {
  uploadStatus: UploadStatus.Error;
  code?: number;
  errorMessage?: string;
}

export type FileInfo = FileInfoWaiting | FileInfoUploadSuccess | FileInfoUploadError;

export type ValidationResult =
  | {
      type: 'success';
    }
  | {
      type: 'error';
      message: string;
    };
