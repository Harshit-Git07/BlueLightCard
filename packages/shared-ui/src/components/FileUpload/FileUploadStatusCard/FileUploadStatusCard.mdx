import { Meta, Canvas } from '@storybook/blocks';
import * as FileUploadStatusCard from './FileUploadStatusCard.stories';

<Meta of={FileUploadStatusCard} />

# FileUpload
- [Overview](#overview)
- [Variants and States](#variants-and-states)
- [Implementation](#implementation)
- [Design](#design)

## Overview
The status card that informs a user of what state their file upload is in.

The visual states are:
- Loading
- Success
- Error

## Variants and States
The Loading State that is shown when a file is being uploaded.
<Canvas of={FileUploadStatusCard.Loading} />

The Success State that is shown after a file has successfully been uploaded.
<Canvas of={FileUploadStatusCard.Success} />

The Error State that is shown after a file failed to upload.
A message can be optionally added.
<Canvas of={FileUploadStatusCard.ErrorWithMessage} />

## Implementation
Provide the following props:
- `fileName` - displays the filename to the user
- `uploadStatus` - determines icon and colours to render
- `removeFile` - callback when the 'x' button is pressed
- `message` - optional message to display errors (can be used for success if needed)

See the *ExampleUploadWrapper* for an example of Implementation.

## Design
[Figma](https://www.figma.com/design/n2NzB4G3OTR74o3dQ4BOwm/My-Account-%26-Sign-up-Component-library?node-id=100-28667&node-type=canvas&t=l1nT8MfPdmiTrK4d-0)
