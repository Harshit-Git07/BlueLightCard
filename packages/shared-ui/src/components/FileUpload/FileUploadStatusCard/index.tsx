import { FC, MouseEvent } from 'react';
import { UploadStatus } from '../types';
import { faSpinner } from '@fortawesome/pro-regular-svg-icons';
import { faCheckCircle, faXmark, faXmarkCircle } from '@fortawesome/pro-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { colours, fonts } from '../../../tailwind/theme';

const getStatusCardIconAndColours = (
  uploadStatus: UploadStatus,
): [IconDefinition, string, string] => {
  switch (uploadStatus) {
    case UploadStatus.Ready:
    case UploadStatus.Loading:
      return [faSpinner, colours.textPrimary, colours.borderOnSurfaceOutline];
    case UploadStatus.Success:
      return [faCheckCircle, colours.textSuccess, colours.borderSuccess];
    case UploadStatus.Error:
      return [faXmarkCircle, colours.textError, colours.borderError];
  }
};

export type Props = {
  fileName: string;
  uploadStatus: UploadStatus;
  removeFile: (event: MouseEvent<HTMLButtonElement>) => void;
  message?: string;
};

const FileUploadStatusCard: FC<Props> = ({ fileName, uploadStatus, removeFile, message }) => {
  const [icon, iconColour, borderColour] = getStatusCardIconAndColours(uploadStatus);
  return (
    <div>
      <div
        className={`p-6 flex gap-2 items-center justify-center border rounded ${colours.backgroundSurfaceContainer} ${borderColour}`}
      >
        <FontAwesomeIcon
          icon={icon}
          className={`h-6 w-6 ${iconColour}`}
          spin={icon === faSpinner}
        />
        <p className={`truncate ${colours.textOnSurface}`}>{fileName}</p>
        <FileUploadCloseButton removeFile={removeFile} />
      </div>
      {message ? <p className={`${iconColour} ${fonts.bodySmall}`}>{message}</p> : null}
    </div>
  );
};

type CloseButtonProps = {
  removeFile: (event: MouseEvent<HTMLButtonElement>) => void;
};

const FileUploadCloseButton: FC<CloseButtonProps> = ({ removeFile }) => (
  <button onClick={removeFile} className="ml-auto flex">
    <FontAwesomeIcon icon={faXmark} className={`h-6 w-6 ${colours.textOnSurfaceSubtle}`} />
  </button>
);

export default FileUploadStatusCard;
