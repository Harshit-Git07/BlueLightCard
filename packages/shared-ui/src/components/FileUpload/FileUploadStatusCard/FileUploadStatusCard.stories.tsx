import { Meta, StoryFn } from '@storybook/react';
import FileUploadStatusCard from '.';
import { UploadStatus } from '../types';

const MaxWidthWrapper: typeof FileUploadStatusCard = (args) => (
  <div className="max-w-[350px]">
    <FileUploadStatusCard {...args} />
  </div>
);

// Meta data of the component to build the story
const componentMeta: Meta<typeof FileUploadStatusCard> = {
  title: 'Component System/FileUpload/StatusCard',
  component: FileUploadStatusCard,
  args: {
    fileName: 'filename.ext',
  },
  parameters: {
    controls: {
      exclude: ['removeFile'],
    },
  },
};

// Define the template which uses the component
const DefaultTemplate: StoryFn<typeof FileUploadStatusCard> = (args) => (
  <MaxWidthWrapper {...args} />
);

export const Loading = DefaultTemplate.bind({});
Loading.args = {
  uploadStatus: UploadStatus.Loading,
};

export const Success = DefaultTemplate.bind({});
Success.args = {
  uploadStatus: UploadStatus.Success,
};

export const ErrorWithoutMessage = DefaultTemplate.bind({});
ErrorWithoutMessage.args = {
  uploadStatus: UploadStatus.Error,
};

export const ErrorWithMessage = DefaultTemplate.bind({});
ErrorWithMessage.args = {
  uploadStatus: UploadStatus.Error,
  message: 'Upload Failed.',
};
ErrorWithMessage.parameters = {
  controls: {
    include: 'message',
  },
};

export const ErrorWithLongMessage = DefaultTemplate.bind({});
ErrorWithLongMessage.args = {
  uploadStatus: UploadStatus.Error,
  message: 'Upload Failed. Upload Failed. Upload Failed. Upload Failed.',
};
ErrorWithLongMessage.parameters = {
  controls: {
    include: 'message',
  },
};

export const LongFileName = DefaultTemplate.bind({});
LongFileName.args = {
  fileName: 'SuperLongFileName.longextension',
  uploadStatus: UploadStatus.Loading,
};
LongFileName.parameters = {
  controls: {
    include: 'fileName',
  },
};

export default componentMeta;
