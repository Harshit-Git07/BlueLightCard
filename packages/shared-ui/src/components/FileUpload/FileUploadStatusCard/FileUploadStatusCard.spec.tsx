import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FileUploadStatusCard, { Props } from '.';
import { UploadStatus } from '../types';
import { userEvent } from '@storybook/testing-library';

const mockRemoveFile = jest.fn();
describe('FileUploadCounter component', () => {
  const defaultProps: Props = {
    fileName: 'testFile.name',
    uploadStatus: UploadStatus.Ready,
    removeFile: mockRemoveFile,
  };

  // smoke test
  it('should render component without error', () => {
    const { baseElement } = render(<FileUploadStatusCard {...defaultProps} />);
    expect(baseElement).toBeTruthy();
  });

  it.each(Object.values(UploadStatus))('matches snapshot for uploadStatus %s', (uploadStatus) => {
    const { container } = render(
      <FileUploadStatusCard {...defaultProps} uploadStatus={uploadStatus} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('triggers callback when clicking close button', async () => {
    render(<FileUploadStatusCard {...defaultProps} />);

    const closeButton = screen.getByRole('button');
    await userEvent.click(closeButton);

    expect(mockRemoveFile).toHaveBeenCalled();
  });
});
