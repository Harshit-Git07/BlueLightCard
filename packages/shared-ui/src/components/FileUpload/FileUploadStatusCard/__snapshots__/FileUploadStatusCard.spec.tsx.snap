// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadCounter component matches snapshot for uploadStatus error 1`] = `
<div>
  <div>
    <div
      class="p-6 flex gap-2 items-center justify-center border rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-error dark:border-colour-error-dark"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-xmark h-6 w-6 text-colour-error dark:text-colour-error-dark"
        data-icon="circle-xmark"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
          fill="currentColor"
        />
      </svg>
      <p
        class="truncate text-colour-onSurface dark:text-colour-onSurface-dark"
      >
        testFile.name
      </p>
      <button
        class="ml-auto flex"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark h-6 w-6 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`FileUploadCounter component matches snapshot for uploadStatus loading 1`] = `
<div>
  <div>
    <div
      class="p-6 flex gap-2 items-center justify-center border rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-spin h-6 w-6 text-colour-primary dark:text-colour-primary-dark"
        data-icon="spinner"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm0 448a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM448 256a32 32 0 1 0 64 0 32 32 0 1 0 -64 0zM32 288a32 32 0 1 0 0-64 32 32 0 1 0 0 64zM75 437a32 32 0 1 0 45.3-45.3A32 32 0 1 0 75 437zm316.8 0A32 32 0 1 0 437 391.8 32 32 0 1 0 391.8 437zM75 75a32 32 0 1 0 45.3 45.3A32 32 0 1 0 75 75z"
          fill="currentColor"
        />
      </svg>
      <p
        class="truncate text-colour-onSurface dark:text-colour-onSurface-dark"
      >
        testFile.name
      </p>
      <button
        class="ml-auto flex"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark h-6 w-6 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`FileUploadCounter component matches snapshot for uploadStatus ready 1`] = `
<div>
  <div>
    <div
      class="p-6 flex gap-2 items-center justify-center border rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-spin h-6 w-6 text-colour-primary dark:text-colour-primary-dark"
        data-icon="spinner"
        data-prefix="far"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288 32a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zm0 448a32 32 0 1 0 -64 0 32 32 0 1 0 64 0zM448 256a32 32 0 1 0 64 0 32 32 0 1 0 -64 0zM32 288a32 32 0 1 0 0-64 32 32 0 1 0 0 64zM75 437a32 32 0 1 0 45.3-45.3A32 32 0 1 0 75 437zm316.8 0A32 32 0 1 0 437 391.8 32 32 0 1 0 391.8 437zM75 75a32 32 0 1 0 45.3 45.3A32 32 0 1 0 75 75z"
          fill="currentColor"
        />
      </svg>
      <p
        class="truncate text-colour-onSurface dark:text-colour-onSurface-dark"
      >
        testFile.name
      </p>
      <button
        class="ml-auto flex"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark h-6 w-6 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;

exports[`FileUploadCounter component matches snapshot for uploadStatus success 1`] = `
<div>
  <div>
    <div
      class="p-6 flex gap-2 items-center justify-center border rounded bg-colour-surface-container-light dark:bg-colour-surface-container-dark border-colour-success dark:border-colour-success-dark"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-circle-check h-6 w-6 text-colour-success dark:text-colour-success-dark"
        data-icon="circle-check"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM369 209L241 337c-9.4 9.4-24.6 9.4-33.9 0l-64-64c-9.4-9.4-9.4-24.6 0-33.9s24.6-9.4 33.9 0l47 47L335 175c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9z"
          fill="currentColor"
        />
      </svg>
      <p
        class="truncate text-colour-onSurface dark:text-colour-onSurface-dark"
      >
        testFile.name
      </p>
      <button
        class="ml-auto flex"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-xmark h-6 w-6 text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          data-icon="xmark"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
